src/js/managers/requester/newsApiFactory.js

```diff
@@ -0,0 +1,45 @@
+import { fetchJSON } from '../../utils/util';
+import proxyLogger from '../logger/proxyLogger';
+
+const newsApiFactory = (method) => {
+  switch (method) {
+    case 'GET':
+      return {
+        execute: url => fetchJSON(url),
+      };
+    case 'POST':
+      return {
+        execute: (url, data) => {
```

Comment:
FYI: each time call of your factory will create new instance with new `execute ` method

I mean that 100 calls of factory for post request will create 100 `execute` functions, not really good way from memory perpective